validate.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash -e
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# "---------------------------------------------------------"
+# "-                                                       -"
+# "-  Validation script checks if Cassandra                -"
+# "-  deployed successfully.                               -"
+# "-                                                       -"
+# "---------------------------------------------------------"
+
+# bash "strict-mode", fail immediately if there is a problem
+set -euo pipefail
+
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck disable=SC1090
+source "$ROOT"/common.sh
+
+APP_MESSAGE="deployment \"hello-server\" successfully rolled out"
+
+# Get the kubectl credentials for the GKE cluster.
+gcloud container clusters get-credentials "$CLUSTER_NAME" --zone "$ZONE" --project "$PROJECT"
+
+# Wait for the rollout of hello-server to finish
+while true
+do
+  ROLLOUT=$(kubectl --namespace default rollout status --watch=false deployment/hello-server) &> /dev/null
```

Comment:
This could loop forever if you don't get APP_MESSAGE. What if it errors instead?

Reply:
Good catch. Let me do one pass validation here.